R_functions/process_estimates_pe.R

```diff
@@ -0,0 +1,186 @@
+  
+  filtered_data <- transformed_pe_data$pe_summarized_effort |> 
+    filter(!is.nan(ang_hrs_mean) & !is.nan(ang_hrs_var) & 
+             !is.nan(est) & !is.nan(var) & !is.nan(l95) & !is.nan(u95))
+  
+  rows_post <- nrow(filtered_data)
+  n_nan <- rows_pre - rows_post
+  
+  #message to user with red text
+  message(paste("NaN values present in the PE effort estimates.", n_nan, "rows filtered out for upload."))
+  # cat("\033[31m", "NaN values present in the PE effort estimates.", n_nan, "rows filtered out for upload.", "\033[39m")
+  cat("\n")
+  
+  #transformation
+  transformed_pe_data$pe_summarized_effort <- transformed_pe_data$pe_summarized_effort |>
+    group_by(analysis_id, project_name, fishery_name, model_type) |>
+    summarise(est_sum = sum(est), .groups = "keep") |> 
+    pivot_longer(cols = c(est_sum),
+                 names_to = "estimate_type",
+                 values_to = "value") |> 
+    #set min date as start of monitoring period
+    mutate(min_event_date = as.Date(params$est_date_start),
+           #set max_event_date as sys.date if in-season, set as est_end_date if out of monitoring period 
+           max_event_date = as.Date(
+             ifelse(
+               Sys.Date() <= params$est_date_end, Sys.Date(),
+               params$est_date_end
+             ))
+    )
+  
+  #tidy output
+  transformed_pe_data$pe_summarized_effort <- transformed_pe_data$pe_summarized_effort |>
+    ungroup() |>
+    mutate(estimate_category = "effort") |>
+    relocate("estimate_category", .after = "model_type")
+
+  # # Effort transformation - !!! ORIGINAL
+  # transformed_pe_data$pe_summarized_effort <- transformed_pe_data$pe_summarized_effort |>
+  #   group_by(analysis_id, project_name, fishery_name, model_type) |>
+  #   summarise(est_sum = sum(est)
+  #   ) |>
+  #   pivot_longer(cols = c(est_sum),
+  #                names_to = "estimate_type",
+  #                values_to = "value") |> 
+  #   #set min date as start of monitoring period
+  #   mutate(min_event_date = as.Date(params$est_date_start),
+  #   #set max_event_date as sys.date if in-season, set as est_end_date if out of monitoring period 
+  #          max_event_date = as.Date(
+  #            ifelse(
+  #              Sys.Date() <= params$est_date_end, Sys.Date(),
+  #              params$est_date_end
+  #              ))
+  #   )
+  # 
+  # #tidy output
+  # transformed_pe_data$pe_summarized_effort <- transformed_pe_data$pe_summarized_effort |>
+  #   drop_na() |>
+  #   ungroup() |>
+  #   mutate(estimate_category = "effort") |>
+  #   relocate("estimate_category", .after = "model_type")
+
+  ### !!! end of effort transformation ####
+    
+  # Catch transformation
+  transformed_pe_data$pe_summarized_catch <- transformed_pe_data$pe_summarized_catch |>
+    group_by(analysis_id, project_name, fishery_name, model_type, est_cg) |> #includes catch group col
+    summarise(n_obs_sum = sum(n_obs),
+              N_days_open_sum = sum(N_days_open),
```

Comment:
Corrected and separated out this calculation as suggested and added to table after catch transformation fe99cd46cda656b2a82c90a1704e1ae5d8e89a62.

Reply:
*actual* correction e447b694615d803275363e03931e741a9099cf62

Reply:
Looks good. Only comment is longer term let's consider making fishery-level metrics like totaldaysopen and totalobs independent of the estimation process so that the user would get those summary values regardless of the model used (PE, BSS, etc.).